internal/testutils/xds/e2e/setup/setup.go

```diff
@@ -59,3 +59,34 @@ func ManagementServerAndResolver(t *testing.T) (*e2e.ManagementServer, string, [
 
 	return xdsServer, nodeID, bc, r
 }
+
+// ManagementServerAndResolverWithSPIFFE sets up an xDS management server, creates
+// bootstrap configuration pointing to that server and creates an xDS resolver
+// using that configuration.
+//
+// Registers a cleanup function on t to stop the management server.
+//
+// Returns the following:
+// - the xDS management server
+// - the node ID to use when talking to this management server
+// - bootstrap configuration to use (if creating an xDS-enabled gRPC server)
+// - xDS resolver builder (if creating an xDS-enabled gRPC client)
```

Comment:
This comment seems to be exactly the same as the one on `ManagementServerAndResolver`. Maybe, you could keep this simple and say that this function is very similar to `ManagementServerAndResolver` except that it uses bootstrap configuration that contains certificate providers that use SPIFFEE certificates.